src/Core/FeatureFlag/FeatureFlagsModifier.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\FeatureFlag;
+
+use Doctrine\ORM\EntityManagerInterface;
+use InvalidArgumentException;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShopBundle\Entity\FeatureFlag;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * This class is in charge of controlling the state modifications of Feature Flags
+ * through the Back Office interface. It fetches the data needed for the Back Office interface
+ * and handles payloads from the Back Office page, replicating the changes on the data model.
+ */
+class FeatureFlagsModifier implements DataConfigurationInterface
+{
+    /** @var EntityManagerInterface */
+    private $doctrineEntityManager;
+
+    /** @var TranslatorInterface */
+    private $translator;
+
+    /**
+     * @param EntityManagerInterface $doctrineEntityManager
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(EntityManagerInterface $doctrineEntityManager, TranslatorInterface $translator)
+    {
+        $this->doctrineEntityManager = $doctrineEntityManager;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfiguration(): array
+    {
+        $configuration = [];
+
+        foreach ($this->getAllFeatureFlags() as $id => $featureFlag) {
+            $configuration[$featureFlag->getName()] = $featureFlag->getState();
```

Comment:
Little suggestion about the `FeatureFlag` entity, I know it's already been merged but it feels weird to use the `getState` for a boolean I would understand if state was an int or a string with more than one value (and it could, we could have a `deprecated` state later for example)

But for a boolean state `isEnabled` would be more adapted I think

Reply:
As you mentioned, `getState()` could become useful and since there is a `state` property I think a getter `getState()` is really useful.

So ... I suggest to get the best of both worlds, and I add a `isEnabled()` method next to `getState()` method. The two are identical but they make sense for a developer. It's [syntaxic sugar](https://en.wikipedia.org/wiki/Syntactic_sugar).

Done in https://github.com/PrestaShop/PrestaShop/pull/23703/commits/bb2f40a2a10a847f5f1144549532bff524cc3074

Reply:
Good idea.